source/treeInterceptorHandler.py

```diff
@@ -13,26 +13,62 @@
 import config
 import braille
 
+#: Identifier for a browse mode setting that
+#: will automatically create tree interceptors for loaded web documents (i.e. not for editable documents.
+TI_CREATE_WEBDOCS = "webDocs"
+#: Identifier for a browse mode setting that
+#: will automatically create tree interceptors for all loaded controls that have a tree interceptor class assigned.
+TI_CREATE_ALWAYS = "always"
+#: Identifier for a browse mode setting that
+#: will never create tree interceptors for all loaded controls that have a tree interceptor class assigned,
+#: unless browse mode is manually activated.
+TI_CREATE_NEVER = "never"
+
+shouldCreateTreeInterceptorOptions = [
+	# Translators: An option in browse mode settings to
+	# always enable browse mode for non editable browse mode documents
+	# (i.e. not for Word documents or aria applications).
+	(TI_CREATE_WEBDOCS, _("For web pages")),
+	# Translators: An option in browse mode settings to
+	# always enable browse mode when a page is loaded.
+	(TI_CREATE_ALWAYS, _("For all supported documents")),
+	# Translators: An option in browse mode settings to
+	# never enable browse mode when a page is loaded.
+	(TI_CREATE_NEVER, _("Never")),
+]
+
 runningTable=set()
 
 def getTreeInterceptor(obj):
 	for ti in runningTable:
 		if obj in ti:
 			return ti
 
-def update(obj):
+def update(obj, force=False):
 	# Don't create treeInterceptors for objects for which NVDA should sleep.
 	if obj.sleepMode:
 		return None
 	#If this object already has a treeInterceptor, just return that and don't bother trying to create one
 	ti=obj.treeInterceptor
 	if not ti:
-		if not obj.shouldCreateTreeInterceptor:
-			return None
 		try:
 			newClass=obj.treeInterceptorClass
 		except NotImplementedError:
 			return None
+		# Import late to avoid circular input.
+		if not force:
+			from browseMode import BrowseModeTreeInterceptor, lastDisableAutoPassThrough
+			if (
+				(obj.shouldCreateTreeInterceptor and config.conf['virtualBuffers']['createTIOnPageLoad'] == TI_CREATE_NEVER) or
+				(not obj.shouldCreateTreeInterceptor and config.conf['virtualBuffers']['createTIOnPageLoad'] != TI_CREATE_ALWAYS) or 
+				# When report pass through on load is on and the tree interceptor class is a browse mode one,
+				# we should not create a tree interceptor if pass through has earlier been enabled explicitly.
+				(config.conf['virtualBuffers']['respectPassThroughOnPageLoad'] and
+					issubclass(newClass, BrowseModeTreeInterceptor) and
+					lastDisableAutoPassThrough
+				)
+			):
+				return None

```

Comment:
Possible improvement: return None is unnecessary. Return works.

Reply:
Update: the style here is return None. You should probably comply with the surrounding code as you are doing.